src/polymer/matchers.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+
+const p = dom5.predicates;
+
+export const isDomBindTemplate =
+    p.AND(p.hasTagName('template'), p.hasAttrValue('is', 'dom-bind'));
+export const isDomModuleTemplate = p.AND(
+    p.parentMatches(p.hasTagName('dom-module')), p.hasTagName('template'));
+export const isTemplate = p.hasTagName('template');
+export const isTemplateDescendant = p.parentMatches(isTemplate);
+
+export interface Template extends parse5.ASTNode { content: parse5.ASTNode; }
+
+export function getNestedDataBindingTemplates(node: parse5.ASTNode) {
```

Comment:
It turns out that this can be written more correctly as a predicate (as top level dom-if and dom-repeat is allowed, it turns out.